### PR TITLE
Control log level with RAILS_LOG_LEVEL in production

### PR DESCRIPTION
**Pull request by dhh**
Makes it easier to switch to debug mode when needed without changing code.

### CODE REVIEW

File: railties/test/application/configuration_test.rb [hunk 1]
```diff
@@ -1787,6 +1787,17 @@ def index
       assert_equal Logger::INFO, Rails.logger.level
     end
 
+    test "config.log_level can be overwritten by ENV['RAILS_LOG_LEVEL'] in production" do
+      restore_default_config
+      app "production"
```
Comment (rafaelfranca) [inline railties/test/application/configuration_test.rb:1792]:
```suggestion
```

File: railties/test/application/configuration_test.rb [hunk 2]
```diff
@@ -1787,6 +1787,17 @@ def index
...
+
+      with_rails_env "production" do
```
Comment (rafaelfranca) [inline railties/test/application/configuration_test.rb:1794]:
```suggestion
```

**Comment by dhh** [comment]
Unrelated CI errors.

**Comment by ioquatix** [comment]
@dhh thanks for this.

Any chance we can change this to `stderr` so we don't mix output? e.g. if someone has a task generating CSV, and log statements are also emitted, you will end up with both of them in the output stream.

**Comment by dhh** [comment]
I'd take a patch to make that toggle easy. Maybe RAILS_LOG_OUTPUT=stderr? But I find it harder for people to get stderr output, and know where to see it, out of the box.

**Comment by ioquatix** [comment]
> But I find it harder for people to get stderr output, and know where to see it

Do you mind elaborating on this problem?

**Pull request merged**